client/src/pages/private/SiteTable.js

```diff
@@ -354,16 +192,47 @@ export default ({ sites, viewOnly }) => {
 
   const openNewPhaseModal = () => {
     closeActionMenu();
-    // TODO HCAP-1277 setActiveModalForm('new-phase');
+    setActiveModalForm('new-phase');
+  };
+
+  const handleFormSubmit = async (submitFunction, formData) => {
+    const response = await submitFunction(formData);
+    if (response.ok) {
+      closeDialog();
+      await fetchSites();
+    }
+    if (response.toast) {
+      openToast(response.toast);
+    }
   };
 
   return (
     <>
-      <SiteFormsDialog
-        activeForm={activeModalForm}
-        onDialogSubmit={fetchSites}
-        onDialogClose={closeDialog}
-      />
+      {activeModalForm === 'new-site' && (
+        <Dialog title={'Create Site'} open={activeModalForm != null} onClose={closeDialog}>
+          <NewSiteForm
+            onSubmit={(values) => {
+              handleFormSubmit(handleSiteCreate, {
+                ...values,
+                siteId: parseInt(values.siteId),
```

Comment:
In the pursuit of cleaning up the markup, can the service method handle parseInt?

then we get a nice `onSubmit={(values) => handleFormSubmit(createSite(values))}`

If we curry `handleFormSubmit` we can simplify this down to

`onSubmit={handleFormSubmit(createSite)}`